### PR TITLE
A few miscellaneous fixes

### PR DESCRIPTION
**Pull request by brryan**
## PR Summary

Caught a few issues and improved a few things in the code, such as:
- `phoedf.py` now no longer requires `g.c.gcov` to be provided if you are in Minkowski geometry
- the `geometry` package now saves the name of the geometry as a parameter
- `cv` wasn't being set quite right in the torus problem initialization
- `b` and `ye` weren't being properly guarded in the radiation source solver failure fixup
- Looped the `status` variable fully through all the rootfinders (e.g. you can still just get a failure status back rather than crashing if you specify bad root bounds)

## PR Checklist

- [x] Adds a test for any bugs fixed. Adds tests for new features.
- [x] Format your changes by calling `scripts/bash/format.sh`.
- [x] Explain what you did.

### CODE REVIEW

File: src/geometry/geometry.cpp [hunk 1]
```diff
@@ -39,6 +39,37 @@ std::shared_ptr<StateDescriptor> Initialize(ParameterInput *pin) {
...
+    geometry_name = "Inchworm";
+  } else if (typeid(PHOEBUS_GEOMETRY) == typeid(Geometry::MonopoleSph)) {
+    geometry_name = "MonopoleSph";
+  } else if (typeid(PHOEBUS_GEOMETRY) == typeid(Geometry::MonopoleCart)) {
+    geometry_name = "MonopoleCart";
+  } else {
+    PARTHENON_THROW("PHOEBUS_GEOMETRY not recognized!");
+  }
+  params.Add("geometry_name", geometry_name);
+
```
Comment (Yurlungur) [inline src/geometry/geometry.cpp:72]: Can we reduce the code here by just letting the preprocessor/cmake handle it? e.g., in `compile_constants.hpp.in` add something like
```C++
#define PHOEBUS_GEOMETRY_STRING "@PHOEBUS_GEOMETRY@"
```
and then we can just do 
```C++
std::string geometry_name = PHOEBUS_GEOMETRY_STRING;
params.add("geometry_name", geometry_name);
```

**Review by Yurlungur — CHANGES_REQUESTED** [review]
Good improvements. I think we can dramatically reduce code and hardcoding in your params thing though

File: src/compile_constants.hpp.in [hunk 2]
```diff
@@ -17,13 +17,12 @@
...
 
 #define PHOEBUS_GEOMETRY Geometry::@PHOEBUS_GEOMETRY@
+#define PHOEBUS_GEOMETRY_STRING "@PHOEBUS_GEOMETRY@"
 #define GEOMETRY_MESH @GEOMETRY_MESH@
 #define GEOMETRY_MESH_BLOCK @GEOMETRY_MESH_BLOCK@
 
 #define USE_VALENCIA @USE_VALENCIA@
 
-#define USE_C2P_ROBUST_FLOORS @USE_C2P_ROBUST_FLOORS@
-
```
Comment (Yurlungur) [inline src/compile_constants.hpp.in:26]: do we not need this anymore?

**Review by Yurlungur — APPROVED** [review]
nice :+1:

File: src/compile_constants.hpp.in (hunk 2 above)
Comment (brryan) [reply src/compile_constants.hpp.in:26]: Nope this is runtime now

File: src/geometry/geometry.cpp (hunk 1 above)
Comment (Yurlungur) [reply src/geometry/geometry.cpp]: Ok I have no idea why my initial idea didn't work, but the following evil preprocessor magic *does* work:
```C++
// in a new file
#define TOKEN2STRING_IMPL(x) #x
#define TOKEN2STRING(x) TOKEN2STRING_IMPL(x)
```
and then:
```C++
// in geometry/coordinate_systems.hpp
constexpr char GEOMETRY_NAME[] = TOKEN2STRING(PHOEBUS_GEOMETRY);
```
which means `GEOMETRY_NAME` will be `Geometry::@PHOEBUS_GEOMETRY` so then we have to pull out the namespace. In `Geomery::Initialize` in `geometry/geometry.cpp` we add this:
``C++
const std::string gname = GEOMETRY_NAME;
const std::string geometry_name = gname.substr(gname.find("::") + 1);
```
and that is the correct name.

File: src/compile_constants.hpp.in [hunk 3]
```diff
@@ -14,18 +14,16 @@
 #ifndef COMPILE_CONSTANTS_HPP_
 #define COMPILE_CONSTANTS_HPP_
 
-#define NCONS_MAX @MAX_NUMBER_CONSERVED_VARS@
+#define NCONS_MAX (@MAX_NUMBER_CONSERVED_VARS@)
```
Comment (Yurlungur) [inline src/compile_constants.hpp.in:17]: I don't think this does anything but I guess it's like slightly more safe stylistically? I got paranoid.

File: src/geometry/coordinate_systems.hpp [hunk 4]
```diff
@@ -24,13 +24,13 @@
 #include <kokkos_abstraction.hpp>
 #include <parthenon/package.hpp>
 #include <utils/error_checking.hpp>
-using namespace parthenon::package::prelude;
```
Comment (Yurlungur) [inline src/geometry/coordinate_systems.hpp:27]: this is dangerous floating around a header not in a namespace.

File: src/geometry/geometry.cpp [hunk 5]
```diff
@@ -39,6 +39,14 @@ std::shared_ptr<StateDescriptor> Initialize(ParameterInput *pin) {
   auto geometry = std::make_shared<StateDescriptor>("geometry");
   Initialize<CoordSysMeshBlock>(pin, geometry.get());
 
+  Params &params = geometry->AllParams();
+
+  // Store name of geometry in parameters
+  const std::string gname = GEOMETRY_NAME;
+  const std::string geometry_name = gname.substr(gname.find("::") + 2);
```
Comment (Yurlungur) [inline src/geometry/geometry.cpp:46]: `find` finds the location of the *first* char in the substring, not the *last*.

File: src/geometry/geometry.cpp [hunk 6]
```diff
@@ -39,6 +39,14 @@ std::shared_ptr<StateDescriptor> Initialize(ParameterInput *pin) {
...
+  PARTHENON_REQUIRE_THROWS(geometry_name.size() > 0, "Invalid geometry name!");
```
Comment (Yurlungur) [inline src/geometry/geometry.cpp:47]: Me being paranoid.

File: src/phoebus_utils/preprocessor_utils.hpp [hunk 7]
```diff
@@ -0,0 +1,7 @@
+#ifndef PHOEBUS_UTILS_PREPROCESSOR_UTILS_HPP_
+#define PHOEBUS_UTILS_PREPROCESSOR_UTILS_HPP_
+
+#define TOKEN2STRING_IMPL(x) #x
+#define TOKEN2STRING(x) TOKEN2STRING_IMPL(x)
```
Comment (Yurlungur) [inline src/phoebus_utils/preprocessor_utils.hpp:5]: Preprocessor tokenization. I have never understood why you need both of these, and can't just do `#x`. But whatever.

File: src/geometry/coordinate_systems.hpp [hunk 8]
```diff
@@ -53,6 +53,7 @@ namespace Geometry {
 // Coordinate system choices
 using CoordSysMeshBlock = GEOMETRY_MESH_BLOCK;
 using CoordSysMesh = GEOMETRY_MESH;
+constexpr char GEOMETRY_NAME[] = TOKEN2STRING(PHOEBUS_GEOMETRY);
```
Comment (Yurlungur) [inline src/geometry/coordinate_systems.hpp:56]: Aside, the existence of this thing means that in the places where we do
```C++
if (typeid(PHOEBUS_GEOMETRY) == typeid(THING))
```
we can instead do
```C++
if constexpr(constexpr_strings_equal(Geometry::GEOMETRY_NAME, "Geometry::XX"))
```
if we implement something like
```C++
#include <string_view>

constexpr bool constexpr_strings_equal(char const * a, char const * b) {
    return std::string_view(a)==b;
}
```
if we're feeling spicy and upgrade to C++17

Comment (brryan) [reply src/geometry/coordinate_systems.hpp:56]: True, something to keep in mind if we revamp geometry in the future (maybe to make it a template parameter so we dont have to recompile everything to get a new geometry)

**Pull request merged**